sys/usb/usbus/msc/msc.c

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2019-2021 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup usbus_msc
+ * @{
+ *
+ * @author  Dylan Laduranty <dylan.laduranty@mesotic.com>
+ * @}
+ */
+
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+#include "usb/usbus/control.h"
+
+#include "usb/msc.h"
+#include "usb/usbus/msc.h"
+#include "usb/usbus/msc/scsi.h"
+#include "board.h"
+
+#include <string.h>
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+#include "mtd.h"
+extern mtd_dev_t *mtd0;
+
+/* Internal buffer to handle size difference between MTD layer and USB
+   endpoint size as some MTD implementation (like mtd_sdcard) doesn't allow
+   endpoint size transfer */
+static unsigned char buff[USBUS_MSC_BUFFER_SIZE];
+
+/* Internal handler definitions */
+static void _event_handler(usbus_t *usbus, usbus_handler_t *handler,
+                           usbus_event_usb_t event);
+
+static int _control_handler(usbus_t *usbus, usbus_handler_t *handler,
+                            usbus_control_request_state_t state,
+                            usb_setup_t *setup);
+
+static void _transfer_handler(usbus_t *usbus, usbus_handler_t *handler,
+                              usbdev_ep_t *ep, usbus_event_transfer_t event);
+
+static void _init(usbus_t *usbus, usbus_handler_t *handler);
+
+static const usbus_handler_driver_t msc_driver = {
+    .init = _init,
+    .event_handler = _event_handler,
+    .control_handler = _control_handler,
+    .transfer_handler = _transfer_handler,
+};
+
+ static const usbus_descr_gen_funcs_t _msc_descriptor = {
+    .fmt_post_descriptor = NULL,
+    .fmt_pre_descriptor = NULL,
+    .len = {
+        .fixed_len =  0,
+    },
+    .len_type = USBUS_DESCR_LEN_FIXED,
+};
+
+static void _write_xfer(usbus_msc_device_t *msc) {
+/* Check if we have a block to read and transfer */
+    if (msc->block_nb) {
```

Comment:
```suggestion
    if (!msc->block_nb) {
        return;
    }
```

It's easier to reason about the code if the level of nesting is kept to a minimum :)

Reply:
Done.